parquet.go

```diff
@@ -32,7 +32,10 @@ func Read[T any](r io.ReaderAt, size int64, options ...ReaderOption) (rows []T,
 		return nil, err
 	}
 	rows = make([]T, file.NumRows())
-	reader := NewGenericReader[T](file, config)
+	reader, err := NewGenericReader[T](file, config)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Changing the signature of `parquet.NewGenericReader` is going to break a lot of applications; however, we can implement this change in a slightly different way.

Instead of passing the `file` as argument to the function, we should first open it with `parquet.OpenFile`, then pass the returned `*parquet.File` to `parquet.NewGenericReader`.

This gives us the opportunity to check of the error returned by `parquet.OpenFile`.

Reply:
That's a very nice idea, I will remove the commit for this change.